test/src/main/java/org/apache/accumulo/test/functional/ThriftMaxFrameSizeIT.java

```diff
@@ -91,14 +91,16 @@ protected abstract class TestMaxFrameSize {
     @Test
     public void testMaxFrameSizeLargerThanDefault() throws Exception {
 
+      int maxSize = TConfiguration.DEFAULT_MAX_FRAME_SIZE;
+      // make sure we go even bigger than that
```

Comment:
I tried to modify this test to use the maxSize of the defaults (so it should end up being the 100MB value), as the basis for this test.

However, that didn't work. I couldn't get past the memory errors.

Reply:
So, on the client side, `ThriftUtil.transportFactory()` is using a hardcoded size on `Integer.MAX_VALUE`. On the server side, with no other properties specified, it's using `GENERAL_MAX_MESSAGE_SIZE` which defaults to `1GB`. I modified the test to set `GENERAL_MAX_MESSAGE_SIZE` to `TConfiguration.DEFAULT_MAX_FRAME_SIZE` and changed `ourSize` to twice that. The tests fails and the following is in the tserver log:

```
2023-08-31T16:44:36,001 [server.TThreadPoolServer] DEBUG: Error processing request
org.apache.thrift.transport.TTransportException: Socket is closed by peer.
        at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:176) ~[libthrift-0.17.0.jar:0.17.0]
        at org.apache.thrift.transport.TTransport.readAll(TTransport.java:100) ~[libthrift-0.17.0.jar:0.17.0]
        at org.apache.thrift.transport.layered.TFramedTransport.readFrame(TFramedTransport.java:132) ~[libthrift-0.17.0.jar:0.17.0]
        at org.apache.thrift.transport.layered.TFramedTransport.read(TFramedTransport.java:100) ~[libthrift-0.17.0.jar:0.17.0]
        at org.apache.thrift.transport.TTransport.readAll(TTransport.java:100) ~[libthrift-0.17.0.jar:0.17.0]
        at org.apache.thrift.protocol.TCompactProtocol.readByte(TCompactProtocol.java:622) ~[libthrift-0.17.0.jar:0.17.0]
        at org.apache.thrift.protocol.TCompactProtocol.readMessageBegin(TCompactProtocol.java:479) ~[libthrift-0.17.0.jar:0.17.0]
        at org.apache.thrift.TMultiplexedProcessor.process(TMultiplexedProcessor.java:104) ~[libthrift-0.17.0.jar:0.17.0]
        at org.apache.accumulo.server.rpc.TimedProcessor.process(TimedProcessor.java:50) ~[classes/:?]
        at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:254) [libthrift-0.17.0.jar:0.17.0]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) [classes/:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) [classes/:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]
```


I'm going to push the changes with this test broken so you can see it.

Reply:
Upon further testing, setting `GENERAL_MAX_MESSAGE_SIZE` is not enough. The other max message size properties have default values. All max message size properties will need to be specified by the user, likely to the same value.

Reply:
IT fixed!